src/deadline_worker_agent/installer/win_installer.py

```diff
@@ -851,18 +866,28 @@ def print_helping_info_and_exit():
         logging.error(f"User does not have Administrator privileges: {os.environ['USERNAME']}")
         print_helping_info_and_exit()
 
+    # Validate that if a windows job user override is specified, that the user exists
     if windows_job_user is not None and not check_account_existence(windows_job_user):
         raise InstallerFailedException(
             f"Account {windows_job_user} provided for argument windows-job-user does not exist. "
             "Please create the account before proceeding."
         )
+    # Validate that if a windows job user override is specified, that it is not the same as the worker agent user
     elif windows_job_user is not None and users_equal(windows_job_user, user_name):
         raise InstallerFailedException(
             f"Argument for windows-job-user cannot be the same as the worker agent user: {user_name}. "
             "If you wish to run jobs as the agent user, set run_jobs_as_agent_user = true in the agent "
             "configuration file."
         )
 
+    # Validate that the --user argument is not a domain user. The installer does not currently support this.
+    if is_domain_user(user_name):
+        raise InstallerFailedException(
+            "running worker agent as a domain user is not currently supported. You can "
+            "have jobs run as a domain user by configuring the queue job run user to specify a "
+            "domain user account."
+        )
+
```

Comment:
This check doesn't occur until after we check if the user is an admin user (and exit if the user isn't an admin). Is that intentional?

Reply:
Good catch. Looks like I reordered the code, but forgot to `pip install ...` before running `install-deadline-worker` and testing E2E again.

I also discovered that the `InstallerFailedException` exceptions were not being caught resulting in unfriendly stack traces being output when these validation exceptions were raised. I fixed that in the new revision